src/inventory_ui.cpp

```diff
@@ -910,7 +915,21 @@ item_location game::inv_map_splice(
                     }
                 }
             }
-            inv_s.make_item_list( slices.back(), pos == g->u.pos() ? &ground_cat : &nearby_cat );
+            if ( match_items_found ) {
+                if( pos == g->u.pos() ) {
+                    inv_s.make_item_list( slices.back(), &ground_cat );
+                } else {
+                    int dist = square_dist( g->u.pos(), pos );
+                    std::string dir_name = direction_name_short( direction_from( g->u.pos(), pos ) );
+                    std::string cat_id = string_format( "NEARBY %d", nearby_rank );
+                    std::string cat_name = string_format( _( "NEARBY %d %s:" ), dist, trim( dir_name ).c_str() );
+
+                    std::unique_ptr<item_category> nearby_cat( new item_category( cat_id, cat_name, nearby_rank ) );
+                    nearby_rank -= 10;
+                    inv_s.make_item_list( slices.back(), nearby_cat.get() );
+                    nearby_cats.push_back( std::move( nearby_cat ) );
```

Comment:
I'd advise against casual use of std::move().  There are optimizations the compiler can make to get move efficiencies without having to explicitly declare it, which avoids the risks it imposes (we've had a half-dozen std::move() based bugs in the short time since switching to c++11).  In other words it should only be used if copying the object in question is demonstrably expensive.
We also need to be a bit careful about emplace methods, some of them aren't well supported on mingw and maybe MSVC.  That one at least causes a compiler error, but only on the affected platforms.